### ISSUE TITLE
Adding neurons and layers to pretrained networks

### ISSUE THREAD

**Issue by kostaleonard** [kostaleonard/dogwood#15]
Can we build off of a pretrained network by adding neurons and layers such that we only improve the model's performance? Based on gradient calculations (see attached notes), we determined that it is possible to add new neurons and layers to a neural network without changing the performance. We also performed initial experiments on adding neurons to pretrained networks. We trained a network on MNIST with a single neuron in the hidden layer, then added a second neuron such that it did not change the network's performance. The new weights from the first dense layer were randomly initialized, but the second dense layer weights were set to 0 to prevent changes to network output. We verified that this would not result in a problem of weight symmetry, then trained the partially-pretrained neural network. We compared the performance of this network to that of one of the same architecture trained from scratch. Performance was approximately equal, with no clear advantage to either training from scratch or using a partially pretrained network. We could investigate this further.

Notebook with experiments available at TODO

![Export0](https://user-images.githubusercontent.com/64159428/155015111-d388ee78-8c11-46f4-a9c7-cdf26b5642f2.png)
![Export1](https://user-images.githubusercontent.com/64159428/155015113-bfb9a517-3f17-4b42-91d3-5a2ab19a4aa8.png)
![Export2](https://user-images.githubusercontent.com/64159428/155015115-7ed8c74a-53ce-4d20-825b-b9f35db53b70.png)
![Export3](https://user-images.githubusercontent.com/64159428/155015116-491a8f9e-e86d-439c-aafb-79ebd39796b7.png)

[mnist.pdf](https://github.com/kostaleonard/dogwood/files/8111641/mnist.pdf)
